lib/y2lan_restart_common.pm

```diff
@@ -148,15 +149,22 @@ sub check_network_status {
         assert_script_run 'dig suse.com|grep \'status: NOERROR\'';    # test if conection and DNS is working
     }
     assert_script_run 'cat journal.log';                              # print journal.log
-    my $is_empty = script_run('[ -s journal.log ]');
-    if ($is_empty) {
-        assert_equals($expected_status, 'no_restart', "Network restart didn't occur, even though it was expected.");
+    my $journal_findings = script_output('cat journal.log');
+    record_info "$expected_status", "Network status expects $expected_status";
+    if ($expected_status eq 'no_restart') {
+        assert_matches(qr/^\s*$/, $journal_findings, "Network service should not have been restarted/reloaded. Check journalctl findings \n$journal_findings\n\n");
```

Comment:
Hmm, 2 things here. First of all such assert is weird, we have checked if file is empty before and I guess it's easier to understand. Second question is if we expect `no_restart`, do we want to accept `reload` then? So I would propose to rename `no_restart` to `no_restart_or_reload`, for example. Then we could simply check that file is empty using `assert_script_run`

Reply:
> First of all such assert is weird, we have checked if file is empty before and I guess it's easier to understand.
the check has been removed. 
> Second question is if we expect no_restart, do we want to accept reload then? So I would propose to rename no_restart to no_restart_or_reload, for example. Then we could simply check that file is empty using assert_script_run

different verifications are happening. for instance `verify_network_configuration(\&check_default_gateway);` does not expect a restart neither a reload. in the other `verify_network_configuration(\&check_network_card_setup_tabs, 'reload');` does expect no_restart but not reload. i think now the expectations are clear and distinguished, unless i miss something.

Reply:
That regex is hard to read, I agree to come back to the original code for that part, I also see feasible this additional state `no_restart_or_reload`, it represents empty file.

Reply:
Sorry now i get the part with the name. yes this can be renamed. more accurate. But i would let the regex as is. it is very basic to check empty strings

Reply:
fyi variable was renamed in previous pull request to avoid to have some unnecessary comment which is missing now saying that empty string means file empty.

Reply:
variable renamed

Reply:
> fyi variable was renamed in previous pull request to avoid to have some unnecessary comment which is missing now saying that empty string means file empty.

thumb up means that want to replace that no so readable regex by the previous code or still want to leave it that way?

Reply:
i thought it was a tip. that you try to give me some part of the big picture. :) for me the regex is quite clear as i have already mentioned.